### PR TITLE
(DOCS-6596) App builder

### PR DESCRIPTION
**Pull request by hestonhoffman**
### Merge instructions

- [x] Please merge after reviewing

### CODE REVIEW

File: config/_default/menus/menus.en.yaml [hunk 1]
```diff
@@ -1448,6 +1448,32 @@ main:
     parent: workflows
     identifier: workflows_connections
     weight: 5
+  - name: App Builder
+    url: service_management/app_builder/
+    pre: AppBuilder
```
Comment (hestonhoffman) [inline config/_default/menus/menus.en.yaml:1453]: @urseberry this is the icon name

File: content/en/service_management/app_builder/build.md [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+for (var item in object) {
+    arr.push(object[item].name);
+}
+
+return arr
+{{< /code-block >}}
+
+#### Post-query hooks
+
+Similar to UI component events, you can configure a reaction to trigger after a query executes. A **Post-query Hook** can set a UI component state, open or close a modal, trigger another query, or even run custom JavaScript. For example, the [ECS Task Manager][7] blueprint's `scaleService` query uses a post-query hook to rerun the `describeService` query after it executes.
```
Comment (cswatt) [inline content/en/service_management/app_builder/build.md:188]:
```suggestion
Similar to UI component events, you can configure a reaction to trigger after a query executes. A **post-query hook** can set a UI component state, open or close a modal, trigger another query, or even run custom JavaScript. For example, the [ECS Task Manager][7] blueprint's `scaleService` query uses a post-query hook to rerun the `describeService` query after it executes.
```

File: content/en/service_management/app_builder/build.md [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+
+#### Error notifications
+
+To display a toast to the user when the system returns an error, toggle **Show Toast on Errors** in the **Advanced** section of a query.
```
Comment (cswatt) [inline content/en/service_management/app_builder/build.md:192]:
```suggestion
To display a toast (a brief message notification) to the user when the system returns an error, toggle **Show Toast on Errors** in the **Advanced** section of a query.
```

i think we really can't assume that people know what "toast" means in the context of UI/UX (i'd actually argue that we also can't assume people know what "toast" means in the context of breakfast foods either)

File: content/en/service_management/app_builder/build.md [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+
+#### Confirmation prompts
+
+To prompt a user for confirmation before the query runs, toggle the **Requires Confirmation** option in the **Advanced** section of a query.
+
+### Variables
+
+Use app variables to pass data from one part of your app to another. Additionally, you can use app variables to pass in data from your dashboard using [dashboard template variables][3].
+
+Variables are enclosed in braces, preceeded by a dollar sign (`${}`). To use a variable, enter the query or UI component name and access the child fields using dot notation. For example, if you have a select component named `select0` and you want to access its default value field, use the syntax `${select0.defaultValue}`. If you're not sure what to enter as a variable, type `${` to open a suggestions menu with all available variables.
```
Comment (cswatt) [inline content/en/service_management/app_builder/build.md:202]:
```suggestion
Variables are enclosed in braces, preceded by a dollar sign (`${}`). To use a variable, enter the query or UI component name and access the child fields using dot notation. For example, if you have a select component named `select0` and you want to access its default value field, use the syntax `${select0.defaultValue}`. If you're not sure what to enter as a variable, type `${` to open a suggestions menu with all available variables.
```

File: content/en/service_management/app_builder/embedded_apps.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+The App Editor modal appears, allowing you to select an app and provide it with a title:
+
+{{< img src="/service_management/app_builder/embedded_apps/app_editor.png" alt="The App Editor modal with an app selected and a widget title" style="width:80%;">}}
+
+## Sync your app with dashboard template variables
+
+You can link your app to template variables anywhere that supports template expressions in your queries or app elements. Use the following code snippet as an example, ensuring to replace the `<TEMPLATE_VARIABLE_NAME>` and `<DEFAULT_VALUE>` with the template variable name and default value, respectively. Paste the snippet into your template expression.
+
+**Note**: If you want to leave an element blank by default (such as a search field), you can set the default value as an empty string `""` or `undefined`.
```
Comment (cswatt) [inline content/en/service_management/app_builder/embedded_apps.md:31]:
```suggestion
**Note**: If you want to leave an element (such as a search field) blank by default, you can set the default value as an empty string `""` or `undefined`.
```

**Review by cswatt — CHANGES_REQUESTED** [review]

File: content/en/service_management/app_builder/_index.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+Datadog App Builder provides an [Action Catalog][1] of over 300 actions across several integrations. The Action Catalog and the connection credentials for each integration are shared with [Datadog Workflow Automation][2]. If there isn't an integration that accomplishes your task, generic actions such as the HTTP request and JavaScript function allow you to perform any task your app requires.
+
+{{< img src="/service_management/app_builder/app-builder-actions.png" alt="Datadog App Builder provides an Action Catalog of over 300 actions across several integrations." style="width:100%;" >}}
+
+## Start with blueprints
+
+Datadog provides you with preconfigured flows in the form of out of the box [blueprints][3] to help you get started.
+
+Below are a few examples of apps you can build:
```
[the remark this answers is not in the document]
Comment (urseberry) [reply content/en/service_management/app_builder/_index.md:29]: I love your improved wording suggestion!

**Pull request merged**